helm-charts/daskhub/values.yaml

```diff
@@ -177,38 +196,118 @@ dask-gateway:
-                    "hub.jupyter.org/username": safe_username,
+                    "hub.jupyter.org/username": user.name,
                     "prometheus.io/scrape": "true",
                     "prometheus.io/port": "8787",
                 }
-                extra_labels = {
-                    "hub.jupyter.org/username": safe_username,
+                worker_extra_pod_annotations = {
+                    "hub.jupyter.org/username": user.name,
                 }
+                picked_ra = ra[options.worker_resource_allocation]
+
                 return {
-                    "worker_cores_limit": options.worker_cores,
-                    "worker_cores": options.worker_cores,
-                    "worker_memory": "%fG" % options.worker_memory,
+                    # A default image is suggested via DASK_GATEWAY__CLUSTER__OPTIONS__IMAGE env variable
                     "image": options.image,
-                    "scheduler_extra_pod_annotations": scheduler_extra_pod_annotations,
                     "scheduler_extra_pod_labels": extra_labels,
+                    "scheduler_extra_pod_annotations": scheduler_extra_pod_annotations,
                     "worker_extra_pod_labels": extra_labels,
+                    "worker_extra_pod_annotations": worker_extra_pod_annotations,
+                    "worker_cores": picked_ra["worker_cores"],
+                    "worker_cores_limit": picked_ra["worker_cores_limit"],
+                    "worker_memory": picked_ra["worker_memory"],
+                    "worker_memory_limit": picked_ra["worker_memory_limit"],
                     "environment": options.environment,
+                    "idle_timeout": options.idle_timeout,
                 }
             return Options(
-                Integer("worker_cores", 2, min=1, label="Worker Cores"),
-                Float("worker_memory", 4, min=1, label="Worker Memory (GiB)"),
-                # The default image is set via DASK_GATEWAY__CLUSTER__OPTIONS__IMAGE env variable
+                Select(
+                    "instance_type",
+                    [it],
+                    default=it,
+                    label="Instance type running worker containers",
+                ),
+                Select(
+                    "worker_resource_allocation",
+                    ra_keys,
+                    default=ra_keys[0],
+                    label="Resources per worker container",
+                ),
+                # The default image is pre-specified by the dask-gateway client
+                # via the env var DASK_GATEWAY__CLUSTER__OPTIONS__IMAGE set on
+                # the jupyterhub user pods
                 String("image", label="Image"),
-                Mapping("environment", {}, label="Environment Variables"),
+                Mapping("environment", {}, label="Environment variables (YAML)"),
+                Integer("idle_timeout", 1800, min=0, label="Idle cluster terminated after (seconds)"),
```

Comment:
Does this value needs to be kept in sync with `c.KubeClusterConfig.idle_timeout` defined below?

Reply:
I don't think it needs, but I figured it made sense. I'll make a note to help them be kept in sync.

Reply:
Thank you!